pumps/resurface_test.go

```diff
@@ -0,0 +1,333 @@
+		"LUVuY29kaW5nOiBjaHVua2VkDQoNCjcNCg0KTW96aWxsYQ0KDQoxMQ0KDQpEZXZlbG9wZXIgTmV0d29yaw0KDQowD" +
+		"QoNCg0KDQo="
+
+	rawRespChunks string = "SFRUUC8xLjEgMjAwIE9LDQpUcmFuc2Zlci1FbmNvZGluZzogY2h1bmtlZA0KQ29udGVudC" +
+		"1UeXBlOiB0ZXh0L2h0bWwNCg0KYw0KPGgxPmdvITwvaDE+DQoNCjFiDQo8aDE+Zmlyc3QgY2h1bmsgbG9hZGVkPC9" +
+		"oMT4NCg0KMmENCjxoMT5zZWNvbmQgY2h1bmsgbG9hZGVkIGFuZCBkaXNwbGF5ZWQ8L2gxPg0KDQoyOQ0KPGgxPnRo" +
+		"aXJkIGNodW5rIGxvYWRlZCBhbmQgZGlzcGxheWVkPC9oMT4NCg0KMA0K"
+
+	rawRespChunksTrailer string = "SFRUUC8xLjEgMjAwIE9LDQpUcmFuc2Zlci1FbmNvZGluZzogY2h1bmtlZA0KQ29" +
+		"udGVudC1UeXBlOiB0ZXh0L2h0bWwNClRyYWlsZXI6IEV4cGlyZXMNCg0KYw0KPGgxPmdvITwvaDE+DQoNCjFiDQo8" +
+		"aDE+Zmlyc3QgY2h1bmsgbG9hZGVkPC9oMT4NCg0KMmENCjxoMT5zZWNvbmQgY2h1bmsgbG9hZGVkIGFuZCBkaXNwb" +
+		"GF5ZWQ8L2gxPg0KDQoyOQ0KPGgxPnRoaXJkIGNodW5rIGxvYWRlZCBhbmQgZGlzcGxheWVkPC9oMT4NCg0KMA0KRX" +
+		"hwaXJlczogV2VkLCAyMSBPY3QgMjAxNSAwNzoyODowMCBHTVQNCg0K"
+)
+
+func SetUp(t *testing.T, url string, queue []string, rules string) (*ResurfacePump, map[string]interface{}) {
+	pmp := ResurfacePump{}
+	cfg := make(map[string]interface{})
+	cfg["capture_url"] = url
+	cfg["queue"] = queue
+	cfg["rules"] = rules
+
+	err := pmp.Init(cfg)
+	if err != nil {
+		t.Fatal("Problem initializing "+pmp.GetName()+": ", err)
+	}
+
+	return &pmp, cfg
+}
+
+func TestResurfaceInit(t *testing.T) {
+	pmp, cfg := SetUp(t, "http://localhost:7701/message", nil, "include debug")
+	assert.NotNil(t, pmp.logger)
+	assert.True(t, pmp.logger.Enabled())
+
+	// Checking with invalid config
+	cfg["capture_url"] = "not a valid URL"
+	pmp2 := ResurfacePump{}
+	err2 := pmp2.Init(cfg)
+	assert.NotNil(t, err2)
+	assert.False(t, pmp2.logger.Enabled())
+}
+
+func TestResurfaceWriteData(t *testing.T) {
+	const MOCK_HOST = "test0"
+
+	pmp, _ := SetUp(t, "", make([]string, 0), "include debug")
+
+	recs := []interface{}{
+		analytics.AnalyticsRecord{
+			Host:         MOCK_HOST,
+			Method:       "GET",
+			ResponseCode: 200,
+			RawRequest:   rawReq,
+			RawResponse:  rawResp,
+			TimeStamp:    time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			Host:         MOCK_HOST,
+			Method:       "POST",
+			ResponseCode: 200,
+			RawRequest:   rawReq,
+			RawResponse:  rawResp,
+			TimeStamp:    time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			Host:         MOCK_HOST,
+			Method:       "GET",
+			ResponseCode: 500,
+			RawRequest:   rawReq,
+			RawResponse:  rawResp,
+			TimeStamp:    time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			Host:         MOCK_HOST,
+			Method:       "Not valid",
+			ResponseCode: 1200,
+			RawRequest:   rawReq,
+			RawResponse:  rawResp,
+			TimeStamp:    time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			Host:        MOCK_HOST,
+			Method:      "GET",
+			RawRequest:  rawReq,
+			RawResponse: rawResp,
+			TimeStamp:   time.Now(),
+		},
+	}
+
+	err := pmp.WriteData(context.TODO(), recs)
+	if err != nil {
+		t.Fatal(pmp.GetName()+"couldn't write records with err:", err)
```

Comment:
we can use assert.NotNil here

Reply:
Thank you, I just modified it accordingly.